### PR TITLE
KAFKA-5278: ConsoleConsumer should honor  `--value-deserializer`

### PR DESCRIPTION
**Pull request by huxihx**
In the original implementation, console-consumer fails to honor `--value-deserializer` config.

### CODE REVIEW

**Comment by huxihx** [comment]
@ijuma Please review the PR. Thanks.

**Comment by huxihx** [comment]
@ijuma Did you get any chances to review this PR? Thanks in advance.

**Comment by huxihx** [comment]
@hachikuji Please review this PR. Thanks.

**Comment by huxihx** [comment]
@ijuma Please take some time to review this PR. Thanks

**Comment by huxihx** [comment]
@guozhangwang A minor change. Please take some time to review. Thanks.

**Comment by guozhangwang** [comment]
LGTM. Thanks for the fix.

**Comment by guozhangwang** [comment]
Merged to trunk and 0.11.0.

File: core/src/main/scala/kafka/tools/ConsoleConsumer.scala [hunk 1]
```diff
@@ -198,8 +198,8 @@ object ConsoleConsumer extends Logging {
     props.putAll(config.extraConsumerProps)
     props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, if (config.options.has(config.resetBeginningOpt)) "earliest" else "latest")
     props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, config.bootstrapServer)
-    props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, if (config.keyDeserializer != null) config.keyDeserializer else "org.apache.kafka.common.serialization.ByteArrayDeserializer")
-    props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, if (config.valueDeserializer != null) config.valueDeserializer else "org.apache.kafka.common.serialization.ByteArrayDeserializer")
+    props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.ByteArrayDeserializer")
+    props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.ByteArrayDeserializer")
```
Comment (ijuma) [inline core/src/main/scala/kafka/tools/ConsoleConsumer.scala:202]: Why is this change needed? Also, we should include tests when fixing bugs. cc @guozhangwang

Comment (ijuma) [reply core/src/main/scala/kafka/tools/ConsoleConsumer.scala:202]: To clarify, it's not clear to me if the consumer config is still respected after this change.

Comment (guozhangwang) [reply core/src/main/scala/kafka/tools/ConsoleConsumer.scala:202]: I think it is respected still: note that the underlying new consumer is always bound with "KafkaConsumer<byte[], byte[]>", so we are actually not serde messages inside consumer ever, but is always relying on the message formatter.

Comment (ijuma) [reply core/src/main/scala/kafka/tools/ConsoleConsumer.scala:202]: Thanks.
